lib/ExecutionEngine/ExecutionEngine.cpp

```diff
@@ -243,6 +244,12 @@ void ExecutionEngine::compile(CompilationMode mode, Function *F,
 
 void ExecutionEngine::compile(Function *F, const CompilationContext &cctx,
                               bool clearOtherFunctions) {
+  // Dump compilation log
+  CompilationScope cscope("ExecutionEngine::compile");
+  if (dumpCompilationLogOpt) {
```

Comment:
Idea: To avoid having this conditional check everywhere, maybe you could move the whole if statement into the constructor of `CompilationScope`?

Reply:
We could also avoid `extern`ing `dumpCompilationLogOpt` if we did this.

Reply:
awesome!